kornia/augmentation/augmentation.py

```diff
@@ -868,10 +975,17 @@ def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor
                                         device=self.device, dtype=self.dtype)
 
     def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
-        return F.compute_crop_transformation(input, params, self.flags)
+        transform: torch.Tensor = get_perspective_transform(
+            params['src'].to(input), params['dst'].to(input))
+        transform = transform.expand(input.shape[0], -1, -1)
+        return transform
 
-    def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
-        return F.apply_crop(input, params, self.flags)
+    def apply_transform(
+        self, input: torch.Tensor, params: Dict[str, torch.Tensor], transform: Optional[torch.Tensor] = None
+    ) -> torch.Tensor:
+        return crop_by_transform_mat(
+            input, transform, self.size,
+            mode=self.resample.name.lower(), padding_mode='zeros', align_corners=self.align_corners)
```

Comment:
Use warp_affine. Besides this function it’s extremely slow. We should allow an alternative based on indexes

Reply:
If think a proper implementation is to detect whether this function is called under torch.no_grad() context. Then to decide the behaviour.